### PR TITLE
Adds RMB click support to clip affordances

### PR DESCRIPTION
**Pull request by crsib**
Resolves: #5195

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Review by vsverchinsky — APPROVED** [review]

File: src/tracks/playabletrack/wavetrack/ui/WaveClipUtilities.h [hunk 1]
```diff
@@ -27,4 +29,6 @@ void fillWhere(
    std::vector<sampleCount>& where, size_t len, bool addBias, double correction,
    double t0, double sampleRate, double stretchRatio, double samplesPerPixel);
 
+std::vector<CommonTrackPanelCell::MenuItem> GetWaveClipMenuItems();
```
Comment (saintmatthieu) [inline src/tracks/playabletrack/wavetrack/ui/WaveClipUtilities.h:32]: I see that this is called from `WaveChannelSubView::GetMenuItems` when clicking the body of a clip, and from `WaveTrackAffordanceControls` when clicking the header of a clip. I wonder if there isn't a way of achieving a simpler flow where whether body or header is clicked, the same code is executed, only with an additional step toto select all clip if the header is hit ?

**Review by saintmatthieu — CHANGES_REQUESTED** [review]
In any case, please use your .clang-format for the new code.
I think moving that new `protected` method to an anonymous namespace is worthwhile trying out, too.
As for the comment about the simpler flow, there I'm much less certain, and would like to hear your thoughts.
Otherwise I tested the behaviour. LGTM just from user experience, but it'd be interesting to check early with @LWinterberg .

File: src/tracks/playabletrack/wavetrack/ui/WaveClipUtilities.h (hunk 1 above)
Comment (crsib) [reply src/tracks/playabletrack/wavetrack/ui/WaveClipUtilities.h:32]: Unfortunately, they are separate, unrelated objects. I want to keep this PR as simple as possible; I don't see much point in complicating stuff here

Comment (crsib) [reply src/tracks/playabletrack/wavetrack/ui/WaveClipUtilities.h:32]: FWTIW this code won't survive the replatforming

File: src/tracks/ui/AffordanceHandle.h [hunk 2]
```diff
@@ -26,5 +26,7 @@ class AUDACITY_DLL_API AffordanceHandle : public TimeShiftHandle
     Result Release(const TrackPanelMouseEvent& event, AudacityProject* pProject, wxWindow* pParent) override;
 
 protected:
+    Result UpdateTrackSelection(const TrackPanelMouseEvent& event, AudacityProject* pProject);
```
[the remark this answers is not in the document]
Comment (crsib) [reply src/tracks/ui/AffordanceHandle.h:29]: It seems that it is no longer needed indeed. I used it when I was trying to get stuff to work

[the remark this answers is not in the document]
Comment (crsib) [reply src/tracks/ui/AffordanceHandle.h:29]: I will move it to `private` then, it still improves readability quite a bit

File: src/tracks/playabletrack/wavetrack/ui/WaveClipUtilities.h (hunk 1 above)
Comment (saintmatthieu) [reply src/tracks/playabletrack/wavetrack/ui/WaveClipUtilities.h:32]: ok

File: src/tracks/ui/AffordanceHandle.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (saintmatthieu) [reply src/tracks/ui/AffordanceHandle.h:29]: Oh, didn't notice. Please check if it could be const, too.

**Review by saintmatthieu — APPROVED** [review]

**Comment by crsib** [comment]
Cosmetics are addressed @saintmatthieu

**Pull request merged**